server/routes/search/searchController.ts

```diff
@@ -0,0 +1,62 @@
+import { Request, Response } from 'express'
+import { PrisonerSearchForm } from 'express-session'
+import HistoricalPrisonerService from '../../services/historicalPrisonerService'
+import logger from '../../../logger'
+import AuditService, { Page } from '../../services/auditService'
+import trimForm from '../../utils/trim'
+import { FindPrisonersByName } from '../../@types/historical-prisoner/historicalPrisonerApiTypes'
+
+export default class SearchController {
+  constructor(
+    private readonly historicalPrisonerService: HistoricalPrisonerService,
+    private readonly auditService: AuditService,
+  ) {}
+
+  async getSearch(req: Request, res: Response): Promise<void> {
+    logger.debug('search /')
+    await this.auditService.logPageView(Page.SEARCH, { who: res.locals.user.username, correlationId: req.id })
+
+    res.render('pages/search', { form: req.session.prisonerSearchForm })
+  }
+
+  async postSearch(req: Request, res: Response): Promise<void> {
+    logger.debug('post search /')
+    // TODO Validate form
+
+    const prisonerSearchForm = req.body
+    req.session.prisonerSearchForm = { ...trimForm(req.body) }
+    logger.debug('searchform is', req.session.prisonerSearchForm)
+
+    if (prisonerSearchForm.searchType === 'name') {
```

Comment:
This looks a bit naff - can change to switch statement - not sure of a neater solution

Reply:
I guess you could move the logic down to the service?